src/main/java/movie/Movie.java

```diff
@@ -0,0 +1,156 @@
+package movie;
+
+import hall.Hall;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Movie {
+
+  private BigDecimal bigDecimal;
+  private String name;
+  private int numberOfRoom;
+  private TypeOfMovie typeOfMovie;
+  private StatusOfMovie statusOfMovie;
+  private LocalDate durationFrom;
+  private LocalDate durationTo;
+  private BigDecimal priceNormal  ;
+  private BigDecimal pricePremium;
+  private BigDecimal priceVip;
+  private Category category;
+  private List<Hall> halls = new ArrayList<>();
+
+  public Movie(String name, int numberOfRoom, TypeOfMovie typeOfMovie,
+      StatusOfMovie statusOfMovie, String durationFrom, String durationTo, double priceNormal,
+      double pricePremium, double priceVip, Category category) {
+
+    this.name = name;
+    this.numberOfRoom = numberOfRoom;
+    this.typeOfMovie = typeOfMovie;
+    this.statusOfMovie = statusOfMovie;
+    this.durationFrom = LocalDate.parse(durationFrom);
+    this.durationTo = LocalDate.parse(durationTo);
+    this.priceNormal = new BigDecimal(priceNormal);
+    this.pricePremium = new BigDecimal(pricePremium);
+    this.priceVip = new BigDecimal(priceVip);
+    this.category = category;
+    //adds to each film a list of places for every day
+    for (int i = 0; i < Duration.between(this.durationFrom.atStartOfDay(), this.durationTo.atStartOfDay()).toDays(); i++) {
```

Comment:
Bad practice to add the loop in the constructor.

Reply:
fixed